### PR TITLE
Feature/run window

### PR DESCRIPTION
**Pull request by dabeshouse**

### CODE REVIEW

File: core/ide/src/main/java/org/alice/stageide/run/RunComposite.java [hunk 1]
```diff
@@ -181,24 +179,21 @@ protected void localize() {
...
-  protected void handlePreShowWindow(Frame frame) {
-    super.handlePreShowWindow(frame);
+  protected void handlePreShowWindow(Frame parentFrame, Frame frame) {
+    super.handlePreShowWindow(parentFrame, frame);
     this.startProgram();
     if (this.size != null) {
       frame.setSize(this.size);
     } else {
-      this.programContext.getOnscreenRenderTarget().getAwtComponent().setPreferredSize(new Dimension(DEFAULT_WIDTH, DEFAULT_HEIGHT));
+      int startingWidth = (int) (parentFrame.getWidth() * 0.9);
```
Comment (weijchen) [inline core/ide/src/main/java/org/alice/stageide/run/RunComposite.java]: can add a variable to define the ratio incase we want to change the ratio in the future

File: core/ide/src/main/java/org/alice/ide/issue/DefaultExceptionHandler.java [hunk 2]
```diff
@@ -187,6 +177,21 @@ private void handleFirstThrowable(Thread thread, Throwable throwable) {
     }
   }
 
+  private Component findOwner() {
+    Application<?> application = Application.getActiveInstance();
+    if (application != null) {
+      if (RunComposite.getInstance().getView().isVisible()) {
+        return RunComposite.getInstance().getView().getAwtComponent();
```
Comment (weijchen) [inline core/ide/src/main/java/org/alice/ide/issue/DefaultExceptionHandler.java]: could be `View view = RunComposite.getInstance().getView()`

**Comment by weijchen** [comment]
some minor suggestions, others lgtm

**Comment by dabeshouse** [comment]
What sort of suggestions? I am only seeing the one comment.

**Comment by weijchen** [comment]
I forgot to submit the reviews...

File: core/ide/src/main/java/org/alice/stageide/run/RunComposite.java (hunk 1 above)
Comment (dabeshouse) [reply core/ide/src/main/java/org/alice/stageide/run/RunComposite.java]: Good idea. Done.

File: core/ide/src/main/java/org/alice/ide/issue/DefaultExceptionHandler.java (hunk 2 above)
Comment (dabeshouse) [reply core/ide/src/main/java/org/alice/ide/issue/DefaultExceptionHandler.java]: Good idea. Done.

**Pull request merged**